### PR TITLE
feat(#50): Tool for routing robots #59

### LINKED ISSUE #50: Tool for routing robots
Tool used to assign a valid route for robots by the player.
Also used to set release/grab zones for the robots.

### PR DESCRIPTION
**Pull request by 42piotrnycz**
closes #50

### CODE REVIEW

File: Assets/Scripts/Robots/RobotRoutingTool.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        [Header("Route")]
+        public List<Route> currentRoute = new List<Route>();
+        public LayerMask floorLayer;
+        public string robotTag;
+
+        [Header("Route Preview")]
+        public bool clearRoutePreviewAfterApplying;
+        private LineRenderer lineRenderer;
+
+
```
Comment (Vie0n) [inline Assets/Scripts/Robots/RobotRoutingTool.cs]:
```suggestion
```

File: Assets/Scripts/Robots/RobotRoutingTool.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        private Vector3 GetLookedAtPointCoordinates()
+        {
+            Ray ray = new Ray(Camera.main.transform.position, Camera.main.transform.forward);
+            RaycastHit hit;
+
+            if (Physics.Raycast(ray, out hit, maxPointDistance, floorLayer))
+            {
+                return hit.point;
+            }
+            else return Vector3.zero;
```
Comment (Vie0n) [inline Assets/Scripts/Robots/RobotRoutingTool.cs]: Redundant
```suggestion
            if (Physics.Raycast(ray, out hit, maxPointDistance, floorLayer))
            {
                return hit.point;
            }
            return Vector3.zero;
```

File: Assets/Scripts/Robots/RobotRoutingTool.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                    if (hit.collider.CompareTag(robotTag))
+                    {
+                        Debug.Log("Switched to new robot: " + hit.collider.gameObject.name);
+                        currentRobot = hit.collider.gameObject;
+                    }
+                }
+            }
+        }
+        public void ApplyRoute(InputAction.CallbackContext context)
+        {
```
Comment (Vie0n) [inline Assets/Scripts/Robots/RobotRoutingTool.cs]: Format all functions to have empty lines between them

File: Assets/Scripts/Robots/Grabber.cs [hunk 4]
```diff
@@ -96,10 +103,17 @@ public void ReleaseItem()
...
+                    if (itemRigidbody is not null)
+                    {
+                        itemRigidbody.isKinematic = false;
+                    }
 
                     Vector3 facingDirection = routeMovement.transform.forward;
                     grabbedItem.transform.position = new Vector3(pos.x, hit.point.y, pos.z) + facingDirection * releaseDistance;
+                    grabbedItem.transform.rotation = Quaternion.identity;
+
+
```
Comment (Vie0n) [inline Assets/Scripts/Robots/Grabber.cs:116]: Remove those

**Comment by 42piotrnycz** [comment]
Made the requested changes, should be fine to merge now.

**Comment by Vie0n** [comment]
LGTM @PatrykPaluch take a look at this too

**Review by Vie0n — APPROVED** [review]

**Review by PatrykPaluch — APPROVED** [review]

**Pull request merged**